pages/card/[id].vue

```diff
@@ -408,6 +412,22 @@ const generateTCGPlayerSearchUrl = (cardName: string): string => {
   return `https://www.tcgplayer.com/search/magic/product?q=${encodedName}`;
 };
 
+function cleanScryfallTcgLink(scryfallLink: string): string {
+  const raw = new URL(scryfallLink).searchParams.get("u");
+  return raw ? decodeURIComponent(raw) : "";
+}
+
+/**
+ * Generate affiliate link for TCGPlayer URLs
+ */
+function getAffiliateLink(tcgUrl: string): string {
+  const cleanUrl = cleanScryfallTcgLink(tcgUrl); // remove any existing affiliate parameter
```

Comment:
This can be replaced by https://github.com/CardMystic/cardmystic/blob/00c91e546e819f90aaa16714f57280eb49f6dac0/models/cardModel.ts#L150C1-L150C39